File: simpliMvvm/src/main/java/com/trayis/simplimvvm/SimpliMvvmApp.java
package com.trayis.simplimvvm;

import android.app.Application;

import com.trayis.simplimvvm.utils.Logging;
import com.trayis.simplimvvm.utils.SimpliMvvmProvider;
import com.trayis.simplimvvm.utils.SimpliProviderUtil;

public abstract class SimpliMvvmApp extends Application {

    @Override
    public void onCreate() {
        super.onCreate();

        initLogger();

        SimpliProviderUtil.setProvider(getSimpliMvvmProvider());
    }

    protected void initLogger() {
        Logging.initLogger();
    }

    protected abstract SimpliMvvmProvider getSimpliMvvmProvider();

}

File: simpliMvvm/src/main/java/com/trayis/simplimvvm/SimpliMvvmApp.kt
package com.trayis.simplimvvm

import android.app.Application
import com.trayis.simplimvvm.utils.SimpliMvvmProvider
import com.trayis.simplimvvm.utils.SimpliProviderUtil

abstract class SimpliMvvmApp : Application() {

    override fun onCreate() {
        super.onCreate()
        SimpliProviderUtil.setProvider(getSimpliMvvmProvider())
    }

    abstract fun getSimpliMvvmProvider(): SimpliMvvmProvider
}